plugins/out_s3/s3.c

```diff
@@ -967,6 +1098,516 @@ static int cb_s3_init(struct flb_output_instance *ins,
+                             in_temp_file);
+    if (bytes == 0) {
+        flb_plg_error(ctx->ins, "GetFileName failed");
+        ret = GetLastError();
+        goto error;
+    }
+
+    bytes = GetTempFileNameA(tmp_path,
+                             TEXT(template_out_prefix),
+                             0, /* create unique name only */
+                             out_temp_file);
+    if (bytes == 0) {
+        flb_plg_error(ctx->ins, "GetFileName failed");
+        ret = GetLastError();
+        goto error;
+    }
+
+    wh = CreateFileA((LPTSTR)in_temp_file,
+                     GENERIC_WRITE,
+                     FILE_SHARE_READ | FILE_SHARE_WRITE | FILE_SHARE_DELETE,
+                     NULL,
+                     CREATE_ALWAYS,
+                     0,
+                     NULL);
+    if (wh == INVALID_HANDLE_VALUE) {
+        ret = -3;
+        goto error;
+    }
+
+    rh = CreateFileA((LPTSTR)out_temp_file,
+                     GENERIC_READ,
+                     FILE_SHARE_READ | FILE_SHARE_WRITE | FILE_SHARE_DELETE,
+                     NULL,
+                     CREATE_ALWAYS,
+                     0,
+                     NULL);
+    if (rh == INVALID_HANDLE_VALUE) {
+        ret = -4;
+        goto error;
+    }
+
+    fdout = _open_osfhandle((intptr_t) rh, _O_RDONLY);
+    if (fdout == -1) {
+        ret = -3;
+        goto error;
+    }
+
+    result = WriteFile(wh, body, body_size, &bytes, NULL);
+    if (!result) {
+        ret = -5;
+        goto error;
+    }
+    CloseHandle(wh);
+
+    ret = build_columnify_command(ctx, in_temp_file, out_temp_file, &parquet_cmd);
+    if (ret != 0) {
+        ret = -1;
+        goto error;
+    }
+
+    cmdp = _popen(parquet_cmd, "r");
+    if (cmdp == NULL) {
+        flb_plg_error(ctx->ins, "command %s failed", DEFAULT_PARQUET_COMMAND_EXISTENCE);
+        return -1;
+    }
+    _pclose(cmdp);
+
+    if (fstat(fdout, &stbuf) == -1) {
+        ret = -6;
+        goto error;
+    }
+    parquet_size = stbuf.st_size;
+    parquet_buf = flb_sds_create_size(parquet_size);
+
+    result = ReadFile(rh, parquet_buf, parquet_size, &bytes, NULL);
+    if (!result) {
+        ret = -5;
+        goto error;
+    }
+
+    CloseHandle(rh);
+
+    if (!DeleteFileA((LPTSTR)in_temp_file)) {
+        ret = -6;
+        flb_plg_error(ctx->ins, "DeleteFileA for %s failed", (LPTSTR)in_temp_file);
+    }
+    if (!DeleteFileA((LPTSTR)out_temp_file)) {
+        ret = -6;
+        flb_plg_error(ctx->ins, "DeleteFileA for %s failed", (LPTSTR)out_temp_file);
+    }
+
+    *payload_buf = parquet_buf;
+    *payload_size = parquet_size;
+
+    flb_sds_destroy(parquet_cmd);
+
+    return 0;
+
+error:
+    if (wh != NULL) {
+        CloseHandle(wh);
+    }
+    if (rh != NULL) {
+        CloseHandle(rh);
+    }
+    if (parquet_cmd != NULL) {
+        flb_sds_destroy(parquet_cmd);
+    }
+
+    return ret;
+}
+
+#else
+static const char *get_tmpdir()
```

Comment:
this is great, but let's also make it configurable

Reply:
I made it be configurable: https://github.com/fluent/fluent-bit/pull/8837/commits/734ad8dde3929a81802487a5280d28c7db2a1f0a